src/main/java/qna/service/QnAService.java

```diff
@@ -35,24 +33,13 @@ public Question findQuestionById(Long id) {
     @Transactional
     public void deleteQuestion(User loginUser, long questionId) throws CannotDeleteException {
         Question question = findQuestionById(questionId);
-        if (!question.isOwner(loginUser)) {
+        List<DeleteHistory> deleteHistories;
+        try {
+            deleteHistories = question.delete(loginUser);
+        } catch (Exception e) {
             throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
         }
```

Comment:
```suggestion
        List<DeleteHistory> deleteHistories  = question.delete(loginUser);
```

CannotDeleteException 을 question 도메인에서 throw 하는게 훨씬 더 깔끔하지 않을까 제안해봅니다.